cpp/include/Ice/MetricsAdminI.h

```diff
@@ -114,28 +107,22 @@ class ICE_API MetricsMapFactory
 
     IceMX::Updater* _updater;
 };
-ICE_DEFINE_PTR(MetricsMapFactoryPtr, MetricsMapFactory);
+using MetricsMapFactoryPtr = std::shared_ptr<MetricsMapFactory>;
 
 template<class MetricsType> class MetricsMapT : public MetricsMapI, private IceUtil::Mutex
 {
 public:
 
-    typedef MetricsType T;
-    typedef ICE_SHARED_PTR<MetricsType> TPtr;
-
-    ICE_DEFINE_PTR(MetricsMapTPtr, MetricsMapT);
+    using T = MetricsType;
+    using TPtr = ICE_SHARED_PTR<MetricsType>;
```

Comment:
Could we update this to `shared_ptr`? Or is this waiting for larger SharedPtr changes?
```suggestion
    using TPtr = std::shared_ptr<MetricsType>;
```

Reply:
No, we need ICE_SHARED_PTR here because MetricsType corresponds to a generated class.